homeassistant/core.py

```diff
@@ -670,6 +670,24 @@ def is_state_attr(self, entity_id, name, value):
         return state_obj is not None and \
             state_obj.attributes.get(name, None) == value
 
+    def get_state_changed(self, entity_id):
```

Comment:
We should not add these to the core state machine class. In fact the `is_state_attr` should probably be moved to template helper too. 

I also don't really get the time changed helpers? If the entity doesn't exist and you get None back, the rest of the template will also start failing. Why not check if it exists before?